crates/glob-match/src/lib.rs

```diff
@@ -0,0 +1,2113 @@
+          // Match escaped characters as literals.
+          if !unescape(&mut c, glob, &mut state.glob_index) {
+            // Invalid pattern!
+            return false;
+          }
+
+          let is_match = if c == b'/' {
+            is_separator(path[state.path_index] as char)
+          } else {
+            path[state.path_index] == c
+          };
+
+          if is_match {
+            state.end_capture(&mut captures);
+
+            if brace_stack.length > 0 && state.glob_index > 0 && glob[state.glob_index - 1] == b'}'
+            {
+              brace_stack.longest_brace_match = state.path_index as u32;
+              state = brace_stack.pop(&state, &mut captures);
+            }
+            state.glob_index += 1;
+            state.path_index += 1;
+
+            // If this is not a separator, lock in the previous globstar.
+            if c != b'/' {
+              state.globstar.path_index = 0;
+            }
+            continue;
+          }
+        }
+        _ => {}
+      }
+    }
+
+    // If we didn't match, restore state to the previous star pattern.
+    if state.wildcard.path_index > 0 && state.wildcard.path_index as usize <= path.len() {
+      state.backtrack();
+      continue;
+    }
+
+    if brace_stack.length > 0 {
+      // If in braces, find next option and reset path to index where we saw the '{'
+      match state.skip_braces(glob, &mut captures, true) {
+        BraceState::Invalid => return false,
+        BraceState::Comma => {
+          state.path_index = brace_stack.last().path_index;
+          continue;
+        }
+        BraceState::EndBrace => {}
+      }
+
+      // Hit the end. Pop the stack.
+      // If we matched a previous option, use that.
+      if brace_stack.longest_brace_match > 0 {
+        state = brace_stack.pop(&state, &mut captures);
+        continue;
+      } else {
+        // Didn't match. Restore state, and check if we need to jump back to a star pattern.
+        state = *brace_stack.last();
+        brace_stack.length -= 1;
+        if let Some(captures) = &mut captures {
+          captures.truncate(state.capture_index);
+        }
+        if state.wildcard.path_index > 0 && state.wildcard.path_index as usize <= path.len() {
+          state.backtrack();
+          continue;
+        }
+      }
+    }
+
+    return negated;
+  }
+
+  if brace_stack.length > 0 && state.glob_index > 0 && glob[state.glob_index - 1] == b'}' {
+    brace_stack.longest_brace_match = state.path_index as u32;
+    brace_stack.pop(&state, &mut captures);
+  }
+
+  !negated
+}
+
+/// gets a slice to a unicode grapheme at the given index
+/// respecting potential escaped characters
+#[cfg(feature = "unic-segment")]
+fn get_char_slice<'a>(glob: &str, glob_bytes: &'a [u8], index: &mut usize) -> Option<&'a [u8]> {
+  use unic_segment::GraphemeCursor;
+  let mut cur = GraphemeCursor::new(*index, glob.len());
+  let end = cur.next_boundary(glob, 0).unwrap().unwrap();
```

Comment:
Traced through and I believe that these unwraps will never get triggered, but just in case could we change these to a helpful panic message?
```suggestion
  let end = cur.next_boundary(glob, 0).unwrap_or_else(|e| panic!("Invalid boundary {:?}", e)).unwrap();
```